code/WorkInProgress/AzrunStuff.dm

```diff
@@ -239,3 +239,353 @@
+	create = 1
+	category = "Component"
+
+/obj/item/organ/brain/ghost // Move to brain.dm
+	name = "Ghost Intelligence Core"
+	desc = "A brain shaped mass of silicon, soulsteel, and LED lights. Attempts to hold onto soul to give life to something else."
+	icon_state = "ghost_brain"
+	item_state = "ai_brain"
+	created_decal = /obj/decal/cleanable/oil
+	made_from = "pharosium"
+	var/activated = 0
+	var/lastTrigger
+	var/datum/movement_controller/ghost_brain/MC
+	var/obj/item/organ/brain/old_brain
+
+	New()
+		..()
+		MC = new
+
+	get_desc()
+		if (usr?.traitHolder?.hasTrait("training_medical"))
+			if (activated)
+				if (src.owner?.key)
+					if (!find_ghost_by_key(src.owner?.key))
+						. += "<br><span class='notice'>[src]'s indicators show that it once had a conciousness captured, but that conciousness cannot be located.</span>"
+					else
+						. += "<br><span class='notice'>[src]'s indicators show that it is still operational, and can be installed into a new body immediately.</span>"
+				else
+					. += "<br><span class='alert'>[src] has powered down fully.</span>"
+			else
+				. += "<br><span class='alert'>[src] is brand new. No conciousness has entered it yet.</span>"
+
+	attack_self(mob/user as mob)
+		if(activated && src.owner?.key && istype(owner.current, /mob/dead/observer))
+			if(alert(user, "Are you sure you want to release the ghost?", "Release Ghost?", "Yes", "No") == "Yes")
+				boutput(owner.current, "You no longer feel anchored to [src]!")
+				owner.current.delStatus("bound_ghost")
+				if(old_brain)
+					owner.brain = old_brain // attempt to restore to previous brain?
+				owner = null
+
+	on_life()
+		var/mob/living/M = holder.donor
+		if(!ishuman(M)) // silicon shouldn't have these problems
+			return
+
+		if(M.client && (isnull(M.client.color) || M.client.color == "#FFFFFF") && !ON_COOLDOWN(src,"ghost_eyes", 5 MINUTES))
+			boutput(M,"Your vision starts to change as your connection this body wavers.")
+			animate(M.client, color=COLOR_MATRIX_GRAYSCALE, time=5 SECONDS, easing=SINE_EASING)
+			animate(color=COLOR_MATRIX_IDENTITY, time=30 SECONDS, easing=SINE_EASING)
+		if(prob(1))
+			boutput(M,"You find you lose control of your body for a moment...")
+			M.changeStatus("paralysis", 2 SECONDS)
+		if(prob(1))
+			boutput(M,"You suddenly feel sluggish as though your connection to your body isn't as strong.")
+			M.changeStatus("slowed", 8 SECONDS, 2)
+
+	get_movement_controller()
+		.= MC
+
+	Crossed(atom/movable/AM)
+		..()
+
+		var/mob/dead/observer/O = AM
+		if(!src.owner && !GET_COOLDOWN(src,"ghost_suck") && istype(O))
+			if(jobban_isbanned(O, "Ghostbrain"))
+				boutput(O, "<span class='notice'>Sorry, you are banned from playing a ghostbrain.</span>")
+				return
+			if(O.can_respawn_as_ghost_critter()) // Azrun TODO New func with more apporpriate verbage?
+				actions.start(new/datum/action/bar/capture_ghost(O), src)
+				ON_COOLDOWN(src, "ghost_suck", 2 SECONDS)
+
+	Entered(atom/movable/A,atom/OldLoc)
```

Comment:
Not the worst way of handling this but note that using `follow thing` as an admin also moves you into a the object. This could possibly mess up the brain as there'd now be two mobs that entered it?